internal/translator/translate.go

```diff
@@ -109,15 +115,40 @@ func translateRoute(irRoute irv1beta1.Route) (hpv1.Route, []string) {
 	return hpRoute, warnings
 }
 
-func translateRoutes(irRoutes []irv1beta1.Route) ([]hpv1.Route, []string) {
+func translateInclude(irRoute irv1beta1.Route) *hpv1.Include {
+	var hpInclude *hpv1.Include
+
+	if irRoute.Delegate != nil {
+		// If there's a delegation, short-circuit.
+		hpInclude = &hpv1.Include{
+			Conditions: []hpv1.Condition{
+				hpv1.Condition{
+					Prefix: irRoute.Match,
+				},
+			},
+			Name:      irRoute.Delegate.Name,
+			Namespace: irRoute.Delegate.Namespace,
+		}
+	}
+
+	return hpInclude
+}
+
+func translateRoutes(irRoutes []irv1beta1.Route) ([]hpv1.Route, []hpv1.Include, []string) {
 
 	var hpRoutes []hpv1.Route
+	var hpIncludes []hpv1.Include
 	var warnings []string
 	for _, irRoute := range irRoutes {
+		hpInclude := translateInclude(irRoute)
+		if hpInclude != nil {
+			hpIncludes = append(hpIncludes, *hpInclude)
+			continue
+		}
 		hpRoute, routeWarnings := translateRoute(irRoute)
-		hpRoutes = append(hpRoutes, hpRoute)
+		hpRoutes = append(hpRoutes, *hpRoute)
```

Comment:
Switching between value and pointer types is a smell that a type declaration is wrong somewhere, why not declare `hpRoutes` as
```
var hpRoutes []*hpv1.Route
```

Reply:
The wrong part here is that `translateRoute()` should return `hpv1.Route` not `*hpv1.Route` - that was a holdover from when I had the Include part in the translateRoute function. Fixed now, if translateRoute gets called, then we'll always get *something* back.